src/lib/microservices/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,342 @@
+    let hints: Array<Hint> = [];
+
+    userConfigs.forEach((userConfig) => {
+        let partialHints = getHintsFromExtends(userConfig.extends);
+
+        partialHints = Object.entries(userConfig.hints).reduce((total: Array<Hint>, [hintName, severity]) => {
+            if (severity === HintStatus.off) {
+                return total;
+            }
+
+            const hintExists = total.some((hint) => {
+                return hint.name === hintName;
+            });
+
+            if (!hintExists) {
+                total.push({
+                    category: categories[hintName],
+                    messages: [],
+                    name: hintName,
+                    status: HintStatus.pending
+                });
+            }
+
+            return total;
+        }, partialHints);
+
+        hints = hints.concat(partialHints);
+    });
+
+    return hints;
+};
+
+/**
+ * Create a new Job in the database.
+ * @param {string} url - The url that the job will be use.
+ * @param {UserConfig} config - The configuration for the job.
+ */
+const createNewJob = async (url: string, configs: Array<UserConfig>, jobRunTime: number): Promise<IJob> => {
+    const hints: Array<Hint> = getHints(configs);
+
+    let databaseJob: IJob;
+
+    try {
+        await database.connect(dbConnectionString);
+        databaseJob = await database.job.add(url, JobStatus.pending, hints, configs, jobRunTime);
+    } catch (e) {
+        logger.error(`Could not connect to databse`, moduleName, e);
+        throw e;
+    }
+
+    return {
+        config: databaseJob.config,
+        error: databaseJob.error,
+        finished: databaseJob.finished,
+        hints: databaseJob.hints,
+        id: databaseJob.id,
+        isNew: true,
+        maxRunTime: databaseJob.maxRunTime,
+        queued: databaseJob.queued,
+        started: databaseJob.started,
+        status: databaseJob.status,
+        url: databaseJob.url,
+        webhintVersion: null
+    };
+};
+
+/**
+ * Get the current active configuration.
+ */
+const getActiveConfig = async (): Promise<IServiceConfig> => {
+    let currentConfig: IServiceConfig;
+
+    try {
+        await database.connect(dbConnectionString);
+        currentConfig = await database.serviceConfig.getActive();
+
+    } catch (e) {
+        logger.error(`Could not connect to databse`, moduleName, e);
+        throw e;
+    }
+
+    if (!currentConfig) {
+        throw new Error('There is no active configuration');
+    }
+
+    const result: IServiceConfig = {
+        active: currentConfig.active,
+        jobCacheTime: currentConfig.jobCacheTime,
+        jobRunTime: currentConfig.jobRunTime,
+        name: currentConfig.name,
+        webhintConfigs: currentConfig.webhintConfigs
+    };
+
+    return result;
+};
+
+/** Create a job to scan an url if it doesn't exist. */
+export const createJob = async (url: string): Promise<IJob> => {
+    /*
+     *   1. Validate input data
```

Comment:
Review the steps, because I think that 1 needs to be reworded and 2 I think is not happening anymore.